File: blog.bak/2011/fibonacci/Fibonacci.java
public class Fibonacci {
  public static int fib(int n) {
    if (n < 2) {
      return n;
    } else {
      return fib(n - 1) + fib(n - 2);
    }
  }

  public static void main(String[] args) {
    System.out.print(fib(40) + "\n");
  }
}

File: blog.bak/2011/fibonacci/Fibonacci.scala
object Fibonacci {
  def fib(n: Int): Int = n match {
    case 0 | 1 => n
    case _ => fib(n -1) + fib(n-2)
  }
  
  def main(args: Array[String]) {
    println(fib(40));
  }
}